File: projectNew/old_src/TTA/TTAOrderListener.java
package com.f14.TTA.listener;

import com.f14.TTA.TTAConfig;
import com.f14.TTA.TTAGameMode;
import com.f14.TTA.TTAPlayer;
import com.f14.TTA.TTAReport;
import com.f14.bg.listener.OrderActionListener;

import java.util.List;

/**
 * TTA的玩家顺序监听器
 *
 * @author F14eagle
 */
public abstract class TTAOrderListener extends OrderActionListener<TTAPlayer, TTAConfig, TTAReport> {


    @Override
    protected List<TTAPlayer> getPlayersByOrder(TTAGameMode gameMode) {
        return gameMode.getGame().getPlayersByOrder(gameMode.getGame().getStartPlayer());
    }
}

File: projectNew/src/com/f14/TTA/listener/TTAOrderListener.kt
package com.f14.TTA.listener

import com.f14.TTA.TTAConfig
import com.f14.TTA.TTAGameMode
import com.f14.TTA.TTAPlayer
import com.f14.TTA.TTAReport
import com.f14.bg.listener.OrderActionListener

/**
 * Created by 吹风奈奈 on 2017/7/19.
 */

abstract class TTAOrderListener(var gameMode: TTAGameMode) : OrderActionListener<TTAPlayer, TTAConfig, TTAReport>(gameMode) {

    override val playersByOrder: List<TTAPlayer>
        get() = gameMode.game.getPlayersByOrder(gameMode.game.startPlayer!!)
}
